src/parser/metrics/RealLinesOfCode.ts

```diff
@@ -16,75 +12,75 @@ let dlog: DebugLoggerFunction = debuglog("metric-gardener", (logger) => {
  * Counts the number of lines in a file, not counting for comments and empty lines.
  */
 export class RealLinesOfCode implements Metric {
-    private commentStatementsSuperSet: QueryStatementInterface[] = [];
+    private commentStatementsSet: Set<string>;
 
     /**
      * Constructs a new instance of {@link RealLinesOfCode}.
      * @param allNodeTypes List of all configured syntax node types.
      */
     constructor(allNodeTypes: ExpressionMetricMapping[]) {
-        allNodeTypes.forEach((expressionMapping) => {
-            if (
-                expressionMapping.metrics.includes(this.getName()) &&
-                expressionMapping.type === "statement"
-            ) {
-                const { expression, activated_for_languages } = expressionMapping;
-                const queryStatement = new ExpressionQueryStatement(
-                    expression,
-                    activated_for_languages
-                );
+        this.commentStatementsSet = new Set(
+            getExpressionsByCategory(allNodeTypes, this.getName(), "comment")
+        );
+    }
 
-                if (expressionMapping.category === "comment") {
-                    this.commentStatementsSuperSet.push(queryStatement);
-                }
+    /**
+     * Waling through the tree in order to find actual code lines.
+     *
+     * Uses a {@link TreeCursor} for this, as according to the
+     * {@link https://tree-sitter.github.io/tree-sitter/using-parsers#walking-trees-with-tree-cursors|Tree-sitter documentation},
+     * this is the most efficient way to traverse a syntax tree.
+     * @param cursor A {@link TreeCursor} for the syntax tree.
+     * @param sureCodeLines A set in which the line numbers of the found code lines are stored.
+     */
+    walkTree(cursor: TreeCursor, sureCodeLines: Set<number>) {
+        // This is not a comment syntax node, so assume it includes "real code".
+        if (!this.commentStatementsSet.has(cursor.currentNode.type)) {
+            // Languages like Go do use new line characters for marking the end of a statement, which leads to wrong
```

Comment:
Would this be something that we could fix in the tree sitter go package? As in: is that something they might also be interested in changing?

Reply:
No, I think that works as intended. While \n has no meaning in other languages and is therefore left out of the syntax tree with the other languages, in Go it seems to be more or less equivalent to a semicolon, so it should be part of the syntax tree.

But as I think of it, since I have removed the inclusion of the last row into the set, I could also remove handling this special case, as it only affected the value of the last row (being one more, due to \n spanning also into the next line).